meson.build

```diff
@@ -554,7 +585,8 @@ pcl = custom_target('prepare_config_last',
              exprecision ? '--exprecision' : [],
             ],
   build_by_default : true,
-  depends: config_h,
+  # kconf isn't needed but won't be generated otherwise
+  depends: [ config_h, conf_kconf ],
```

Comment:
This should be a dependency where configh is used. Maybe a dep declaration.

Reply:
Hm, I'm getting an error:
```
meson.build:532:17: ERROR: Failed to load /Users/mateusz/CLionProjects/HaoZekeOpenBLAS/build/config.kconf: [Errno 2] No such file or directory: '/Users/mateusz/CLionProjects/HaoZekeOpenBLAS/build/config.kconf'
```

It looks like `conf_kconf` isn't present in the `setup` step (I think it's created in the `compile` step) so:
```
conf_kv = keyval.load(meson.current_build_dir() / 'config.kconf')
```
is missing the file.

Reply:
Yup, sorry, could you push a fix? Am commuting.. I think it needs to be a dependency on something before the file is present, similar to the way config_h is handled.

Reply:
I'm not sure if this can be done: `config.h` is present only in the `meson compile` step. The dictionary building with `'keyval'` is strictly a `meson setup` step I think, it can't be done lazily. So `config.kconf` will never be present when `keyval.load(...)` is called.

Reply:
Makes sense. What about fs? The python script can generate the file in the source root via an argument and if fs is at compile time that should work

Reply:
Hmm I think meson's `fs` module operations are also `setup` step. If you mean to create a separate custom target which does this key-value file generation with a Python script in the `compile` step then I wonder about another issue:

`dir` and `kernel` selection is also a `setup` step. So I wonder how to move this selection to the `compile` step based on subarchitecture identified in the mentioned key-value file: 
```
'_r': {'dir': 'x86_64', 'kernel': 'cgemm_kernel_8x2_haswell.c', 'addl': ['-DNC']},
```

Even if possible, this looks really hacky to me. It should be all set in stone after `meson setup`, but this again requires going back to the issue of `config.h` be present in the `meson setup` step.

Reply:
Yeah it's not clean conceptually. I guess this is then waiting on the other branch where I'm working on config (ah I don't think I pushed it yet).

For now if you comment it out and get a config.h then run it with it uncommented you'll be able to populate the configuration data to test architectures.